### PR TITLE
Define metadata in setup.cfg

### PR DESCRIPTION
**Pull request by nbraud**
- [x] Move version number to `/version` file, define `__version__` based on package metadata
  That file doesn't need to be included in the built package, but stays separate for Cirrus to pick up on.
- [x] Move author definition to `setup.cfg`
- [x] Drop setup-time dependencies
  We aren't using `attr:` anymore in `setup.cfg`, so `emanate` needs not be importable at setup-time.

I'm much happier with this; there's presumably a bit of import-time overhead, but
it's worth the reduction in complexity.

### CODE REVIEW

**Comment by nbraud** [comment]
@duckinator There might be complications when merging this, since it will match the `changesInclude('version')` condition in Cirrus; does the release task fail cleanly if a release with that number already exists?

**Comment by nbraud** [comment]
Oh, `importlib.metadata` was introduced in Python 3.8  >_>'
I added a conditional dependency on a polyfill (by the CPython maintainers, unlike the `dataclasses` mess)

**Comment by nbraud** [comment]
Oh FFS.  Now it barfs up because of a dependency conflict with `bork`.

@duckinator, can we move away from specifying exact versions?

**Comment by nbraud** [comment]
@AstraLuma, I'd be interested in your opinion  :purple_heart:

File: emanate/__init__.py [hunk 1]
```diff
@@ -15,14 +15,14 @@
...
-__author__ = "Ellen Marie Dash"
-
-# __version__ is defined in version.py.
+try:
+    from importlib import metadata
+except ImportError:
+    import importlib_metadata as metadata  # type: ignore
+
+# Set the module-level dunders suggested in PEP8
+__author__ = metadata.metadata('emanate').get('author')
```
Comment (nbraud) [inline emanate/__init__.py:24]: Is there a way to pass `metadata` the current module, instead of hardcoding a name?

**Comment by nbraud** [comment]
Annoyingly, we presumably care about OpenSUSE (at least, @duckinator included it in [versions.boop.ws]) and even their current beta still ships Python 3.6  :sob: 

[versions.boop.ws]: https://versions.boop.ws

**Comment by duckinator** [comment]
@nbraud I can't really give this a proper review because GitHub is incomprehensible on a 1366x768 display, but:

1. the release task should fail cleanly.
2. ok @ the `importlib.metadata` polyfill. this will make the zipapp Py3.8+ unless we make it unconditional, but i'm not terribly worried about that unless people actively ask for 3.7 support in the zipapp.
3. Bork is already moving away from pinning exact versions, and I am okay with doing the same for Emanate.
4. i don't care about OpenSUSE unless people are trying to use one of my projects on it.

**Comment by nbraud** [comment]
> 1. the release task should fail cleanly.

OK, then there should be no special handling needed for this PR (moving the `version` file).

> 2. ok @ the `importlib.metadata` polyfill. this will make the zipapp Py3.8+ unless we make it unconditional, but i'm not terribly worried about that unless people actively ask for 3.7 support in the zipapp.

Ayup.  I'm not terribly-worried either about that polyfill.
Ideally, we could make the dependency conditional on something like `python_version < '3.8' or building_as_zipapp` but there seem to be no way to express that in the [PEP 508 language](https://www.python.org/dev/peps/pep-0508/#environment-markers).

> 3. Bork is already moving away from pinning exact versions, and I am okay with doing the same for Emanate.

OK.  Let's do another PR afterwards to un-pin all our dependencies then.

> 4. i don't care about OpenSUSE unless people are trying to use one of my projects on it.

Good to know.  Given that I don't think we'd want to drop Py3.7 compatibility just yet, we need to deal with the polyfills anyway (and edge-cases like zipapps) so I guess we could keep Py3.6 compatibility so long as the `dataclasses` shim doesn't get in the way; on the other hand, it's in my experience a problematic library (badly maintained, had bugs that had been fixed in the Python stdlib for years, ...) so I'd be happy not to have a dependency on it, even a conditional one.

**Comment by nbraud** [comment]
Modified to use the `~=` version specifier, since that will (presumably) be what unpinned dependencies will use.

File: emanate/__init__.py (hunk 1 above)
Comment (AstraLuma) [reply emanate/__init__.py:24]: `__name__`

**Comment by duckinator** [comment]
@nbraud This looks good to me &mdash; if you're happy with it, go ahead and merge it when you're ready.

**Comment by nbraud** [comment]
bors r=duckinator

**Comment by nbraud** [comment]
nors abort

**Comment by nbraud** [comment]
bors abort

**Comment by nbraud** [comment]
Oops I had forgotten to push a commit  >_>'

**Comment by nbraud** [comment]
bors r=duckinator

**Pull request merged**